java-squid/src/main/java/org/sonar/java/se/ExplodedGraphWalker.java

```diff
@@ -235,8 +234,14 @@ private void handleBlockExit(ExplodedGraph.ProgramPoint programPosition) {
       }
     }
     // unconditional jumps, for-statement, switch-statement, synchronized:
-    for (CFG.Block successor : block.successors()) {
-      enqueue(new ExplodedGraph.ProgramPoint(successor, 0), programState);
+    if(node.exitPath) {
+      enqueue(new ExplodedGraph.ProgramPoint(block.exitBlock(), 0), programState, true);
+    } else {
+      for (CFG.Block successor : block.successors()) {
+        if (!block.isFinallyBlock() || successor != block.exitBlock()) {
+          enqueue(new ExplodedGraph.ProgramPoint(successor, 0), programState, successor == block.exitBlock());
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/coding_rules#rule_key=squid%3AS1698)